pkgs/applications/office/gnucash/default.nix

```
@@ -1,106 +1,210 @@
-, boost, icu, libxml2, libxslt, gettext, swig, isocodes, gtk3, glibcLocales
-, webkitgtk, dconf, hicolor-icon-theme, libofx, aqbanking, gwenhywfar, libdbi
-, libdbiDrivers, guile, perl, perlPackages
+{
+  # nix "native" dependencies
+  fetchurl
+, lib
+, stdenv
+
+, glibcLocales  # necessary to pass locale-related GnuCash tests
+, wrapGAppsHook # nix sugar for Gnome apps
+
+, makeWrapper   # used to wrap gnucash with Gnome sugar and the cli utils which need Perl
+
+, cmake         # tell nix to use cmake rather than make
+
+  # GnuCash requirements
+, aqbanking
+, boost
+, glib
+, gtest
+, guile
+, gwenhywfar
+, icu
+, libdbi
+, libdbiDrivers
+, libofx
+, libxml2
+, libxslt
+, perl
+, pkg-config
+, swig
+, webkitgtk
+
+  # GnuCash cli utils' Perl requirements
+, perlPackages
 }:
 
-let
-
-  # Enable gnc-fq-* to run in command line.
-  perlWrapper = stdenv.mkDerivation {
-    name = perl.name + "-wrapper-for-gnucash";
-    nativeBuildInputs = [ makeWrapper ];
-    buildInputs = [ perl ] ++ (with perlPackages; [ FinanceQuote DateManip ]);
-    dontUnpack = true;
-
-    installPhase = ''
-      mkdir -p $out/bin
-      for script in ${perl}/bin/*; do
-        makeWrapper $script $out''${script#${perl}} \
-          --prefix "PERL5LIB" ":" "$PERL5LIB"
-      done
-    '';
-  };
-
-in
-
 stdenv.mkDerivation rec {
-  pname = "gnucash";
+  pname   = "gnucash";
   version = "4.9";
 
   src = fetchurl {
-    url = "mirror://sourceforge/gnucash/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-mlUcMMG3EhmfwiJ6EJr7mE177xjhOBcLvHIlxsH6ty0=";
+    sha256 = "0bdpzb0wc9bjph5iff7133ppnkcqzfd10yi2qagij4mpq4q1qmcs";
+    url    = "https://github.com/Gnucash/gnucash/releases/download/${version}/gnucash-${version}.tar.bz2";
   };
 
-  nativeBuildInputs = [ pkg-config makeWrapper cmake gtest swig ];
-
-  buildInputs = [
-    boost icu libxml2 libxslt gettext isocodes gtk3 glibcLocales
-    webkitgtk dconf libofx aqbanking gwenhywfar libdbi
-    libdbiDrivers guile
-    perlWrapper perl
-  ] ++ (with perlPackages; [ FinanceQuote DateManip ]);
-
-  propagatedUserEnvPkgs = [ dconf ];
-
-  # glib-2.62 deprecations
-  NIX_CFLAGS_COMPILE = "-DGLIB_DISABLE_DEPRECATION_WARNINGS";
-
-  # this patch disables test-gnc-timezone and test-gnc-datetime which fail due to nix datetime challenges
-  patches = [ ./0001-changes.patch ];
-
-  postPatch = ''
-    patchShebangs .
+  nativeBuildInputs = [
+    glibcLocales
+    wrapGAppsHook
+
+    cmake
+
+    makeWrapper
+
+    aqbanking
+    boost
+    glib
+    gtest
+    guile
+    gwenhywfar
+    icu
+    libdbi
+    libdbiDrivers
+    libofx
+    libxml2
+    libxslt
+    perl
+    pkg-config
+    swig
+    webkitgtk
+  ];
+
+  # GnuCash cli utils' Perl requirements
+  propagatedBuildInputs = with perlPackages; [
+    DateManip
+    FinanceQuote
+  ];
+
+  patches = [ ./0001-changes.patch ]; # this patch disables test-gnc-timezone and test-gnc-datetime which fail due to nix datetime challenges
```

```suggestion
  patches = [
    # disables test-gnc-timezone and test-gnc-datetime which fail due to nix datetime challenges
    ./0001-changes.patch
  ];
```
patch comments always go before the patch the belong to

---

fixed in 072900f31f1598aca35feb66e4c553bbe69a9509